nservicebus/messaging/databus/index.md

```diff
@@ -95,16 +95,18 @@ Automatically removing these attachments can cause problems in many situations.
 
 ## Alternatives
 
-- Use a different transport or a different transport tier
-- Message Compression: Use message body compression which works well on text-based payloads like XML and Json or any payload (text or binary) that contains repetitive data
-  - [Message mutator example demonstrating message body compression](/samples/messagemutators/)
-- Stream-based properties: The [Handling large stream properties via pipeline](/samples/pipeline/stream-properties/) sample demonstrates a purely stream-based approach (rather than loading the full payload into memory) implemented by leveraging the NServiceBus pipeline.
-- Binary Serializer: A binary serializer is more efficient and most serializers can be added with a few lines of code
-   - Some binary [serializers are maintained by the community](/nservicebus/community/#serializers)
-- Attachments: When dealing with unbounded binary payloads consider the [community maintained NServiceBus.Attachments](/nservicebus/community/#nservicebus-attachments)
-  - Read on demand: Will only retrieve attachment data when the consumer reads it
-  - Reduced Memory usage: No base64 serializer overhead resulting in a significant reduction in resource utilization
-- Use any of the above in combination with compression
+NOTE: A combination of these techniques may be used.
+
+- Use a different transport or different tier (e.g. Azure Service Bus _Premium_ instead of _Standard_).
+- Message compression: This works well on text-based payloads like XML and JSON, or any payload (text or binary) that contains repetitive data.
+  - See the [message mutator example demonstrating message body compression](/samples/messagemutators/).
+- Stream-based properties: The sample showing [handling large stream properties via pipeline](/samples/pipeline/stream-properties/) demonstrates a purely stream-based approach (rather than loading the full payload into memory) using the NServiceBus pipeline.
+- A more efficient serializer: For example, a binary serializer. Most serializers can be added with a few lines of code.
+   - Several [serializers are maintained by the community](/nservicebus/community/#serializers).
+- Attachments: The community maintained [NServiceBus.Attachments](/nservicebus/community/#nservicebus-attachments) package may be used for unbounded binary payloads. The package is similar to the Data Bus but has some differences:
+  - Read on demand: Attachments are only retrieved when read by a consumer.
+  - Async enumeration: The package supports processing all data items using an `IAsyncEnumerable`.
+  - Reduced memory usage: The package avoids the use of a base64 serializer which results in a significant reduction in memory usage.
 
```

Comment:
@ramonsmits this is your suggestion from https://github.com/Particular/docs.particular.net/pull/6199#discussion_r1336862452 —

```suggestion
  - Better suited for [binary large objects (BLOBs](https://en.wikipedia.org/wiki/Binary_large_object) as it does not have (de)serialization overhead.

```

I'm not sure I understand and nor will a potential reader. This difference to the databus is [not listed in the package README](https://github.com/NServiceBusExtensions/NServiceBus.Attachments#compared-to-the-databus). Can you please explain more about why this package is better suited for BLOBS?

From the README I see:

> With the DataBus the only interaction is via byte arrays. NServiceBus.Attachments supports reading and writing using streams, byte arrays, or string.

How does this make it better for BLOBs?

Reply:
The databus stores byte arrays (as everything does...) but handles all this in memory and (de)serializes properties to/from byte arrays which is different with NServiceBus.Attachments

If you actually have BLOB's like large files NServiceBus.Attachments is by far the better option.

Reply:
> which is different with NServiceBus.Attachments

How is it different? What does NServiceBus.Attachments do?

> If you actually have BLOB's like large files NServiceBus.Attachments is by far the better option.

Sorry, maybe I'm being slow, but I still don't understand why.

Reply:
> Sorry, maybe I'm being slow, but I still don't understand why.

Well, maybe then read documentation on how streams work?

Reply:
@adamralph WHen using streams and you are purely processing binary BLOB's like PDF's, image, video, backup files, logs,e tc

a. Things don't need to be fully loaded in memory. These BLOBs are likely being read from (relatively slow) file/network streams anyway to just passing data directly without requiring to load these in memory.
b. Don't require (de)serialization by very inefficiently loading binary data, then serializing that to Base64 for storage.

Reply:
@ramonsmits thanks, I think I get it now. This seems to tie in with the previous bullet, so I'll make a suggestion on that.

Reply:
_continued in https://github.com/Particular/docs.particular.net/pull/6199/files#r1372717404_